### ISSUE TITLE
torch.zeros and other factory functions don't work with no arguments

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#6739]
Their argument specification is `torch.zeros(*sizes)`; thus, by induction, `torch.zeros()` should give you a zero scalar (since sizes=()). However, this is currently rejected.

**Comment by gchanan** [comment]
numpy behaves the same way, although they list the argument type as "int or sequence of ints".

**Comment by apaszke** [comment]
It's also a bit ambiguous. Should it return a zero scalar or an empty tensor?

**Comment by ezyang** [comment]
@apaszke It's not ambiguous, if I pass `sizes=()` I should get a scalar. It's not the end of the world if we don't make this work but I don't like having to special case in the 0-dim case.

**Comment by ezyang** [comment]
closing as WONTFIX; uses the explicit `sizes` syntax if you need the inducitve case.
